src/vhost_user/slave_req_handler.rs

```diff
@@ -610,27 +606,24 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
     fn set_slave_req_fd(
         &mut self,
         hdr: &VhostUserMsgHeader<MasterReq>,
-        rfds: Option<Vec<RawFd>>,
+        files: Option<Vec<File>>,
     ) -> Result<()> {
-        if let Some(fds) = rfds {
-            if fds.len() == 1 {
-                let sock = unsafe { UnixStream::from_raw_fd(fds[0]) };
-                let vu_req = SlaveFsCacheReq::from_stream(sock);
-                self.backend.set_slave_req_fd(vu_req);
-                self.send_ack_message(&hdr, Ok(()))
-            } else {
-                Err(Error::InvalidMessage)
-            }
-        } else {
-            Err(Error::InvalidMessage)
+        let mut files = files.ok_or(Error::InvalidMessage)?;
```

Comment:
How about adding a helper function to extract the only one fd from Option<Vec<File>>?

Reply:
Good idea. I added a helper function [`take_single_file()`](https://github.com/rust-vmm/vhost/pull/39/files#diff-7ac2f087f60290ddeaa6195f8dcb4fc3cace310c7c84b31cbbc97174832fedf7R768).